src/averageAges.js

```diff
@@ -15,11 +15,17 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let males = people.filter(person => person.sex === 'm');
+
+  if (century) {
+    males = males.filter(person => Math.ceil(person.died / 100) === century);
+  }
+
+  const averageMaleAge = males
+    .map(person => person.died - person.born)
+    .reduce((sum, x) => sum + x) / males.length;
```

Comment:
також просив спробувати зробити без мапу
```suggestion
  const averageMaleAge = males
    .reduce((sum, person) => sum + (person.died - person.born)), 0) / males.length;
```